src/wallet/rpcwallet.cpp

```diff
@@ -216,7 +216,7 @@ static void SetFeeEstimateMode(const CWallet& wallet, CCoinControl& cc, const Un
         if (!estimate_mode.isNull() && estimate_mode.get_str() != "unset") {
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot specify both estimate_mode and fee_rate");
         }
-        cc.m_feerate = CFeeRate(AmountFromValue(fee_rate), COIN);
+        cc.m_feerate = CFeeRate::FromSatB(AmountFromValue(fee_rate));
```

Comment:
In commit "wallet: update rpcwallet to CFeeRate named constructors" (c72083c2d56237f8fbd3a82e3ff395f0cdb8fd57)

Wow wow. I see that this is doing the right thing, but these constructor names seem absurdly misleading! The `AmountFromValue` function converts a floating point number to a fixed-point representation of that number multiplied by `10e8`. So even though the `fee_rate` univalue argument has units of sat/B, the `AmountFromValue` integer return value is no longer in the same units, because it has been multiplied by 100000000. So `FromSatB` should really be called `CFeeRate::FromSatB_Fixed_Point_e8` or something like that because it is taking sat/B values with a very specific, atypical number representation.

An analogous thing is happening with `FromBtcKb` below. It doesn't really take a BTC/Kb argument. That's basically false advertising. What it really takes is a BTC/Kb*10e8 argument which is the same as a sat/Kb argument, so this constructor would be more accurately called `CFeeRate::FromSatKb`

I guess I'm having a negative reaction to these constructor names, but I do think this could be cleaned up pretty easily. I think it would be better if `CFeeRate` class had no involvement with univalue parsing or fixed point representations, and if it didn't (mis)use the CAmount type as a way to represent rates when `CAmount` is used almost everywhere else only to represent quantities.

My suggestion to be less confusing / misleading would be to drop the two new `CFeeRate` constructors, and just add standalone RPC util functions:

```c+++
CFeeRate FeeRateFromSatB(const UniValue& value);
CFeeRate FeeRateFromBtcKb(const UniValue& value);
```

Then call these to simplify `CFeeRate::FromSatB(AmountFromValue(feerate))` as `FeeRateFromSatB(feerate)` here and in `setfeerate` and to simplify `CFeeRate::FromBtcKb(AmountFromValue(feerate))` as `FeeRateFromBtcKb(feerate)` in `settxfee`

Reply:
> Wow wow. I see that this is doing the right thing, but these constructor names seem absurdly misleading!

The current code in master isn't exactly clear right now either:

```cpp
    /** Constructor for a fee rate in satoshis per kvB (sat/kvB).
     *
     *  Passing a num_bytes value of COIN (1e8) returns a fee rate in satoshis per vB (sat/vB),
     *  e.g. (nFeePaid * 1e8 / 1e3) == (nFeePaid / 1e5),
     *  where 1e5 is the ratio to convert from BTC/kvB to sat/vB.
     */
    CFeeRate(const CAmount& nFeePaid, uint32_t num_bytes);
```

I think it would be good to figure out how to clear this up before introducing further features that depend on this.

Reply:
Original discussion for reference: https://github.com/bitcoin/bitcoin/pull/20305#discussion_r522371491

Reply:
> > Wow wow. I see that this is doing the right thing, but these constructor names seem absurdly misleading!
> 
> The current code in master isn't exactly clear right now either:
> 
> ```c++
>     /** Constructor for a fee rate in satoshis per kvB (sat/kvB).
>      *
>      *  Passing a num_bytes value of COIN (1e8) returns a fee rate in satoshis per vB (sat/vB),
>      *  e.g. (nFeePaid * 1e8 / 1e3) == (nFeePaid / 1e5),
>      *  where 1e5 is the ratio to convert from BTC/kvB to sat/vB.
>      */
>     CFeeRate(const CAmount& nFeePaid, uint32_t num_bytes);
> ```
> 
> I think it would be good to figure out how to clear this up before introducing further features that depend on this.

Ok, seems best to close this then. It's almost a year old and needs to be done differently.  I think using sat/vB is still a popular request from users and plan to try again a bit later.